bot/extensions/polls/views.py

```diff
@@ -0,0 +1,102 @@
+import discord
+from discord import ui
+
+
+class PollModal(ui.Modal):
+    name = ui.TextInput(label="Choice name", placeholder="Enter poll choice", max_length=50, required=True)
+
+    def __init__(self, var: discord.Interaction):
+        self.var = var
+        super().__init__(title="Poll options")
+
+    @property
+    def reactions(self):
+        return {
+            0: "1️⃣",
+            1: "2️⃣",
+            2: "3️⃣",
+            3: "4️⃣",
+            4: "5️⃣",
+            5: "6️⃣",
+            6: "7️⃣",
+            7: "8️⃣",
+            8: "9️⃣",
+            9: "🔟",
+        }
+
+    async def on_submit(self, interaction: discord.Interaction) -> None:
+        message = await self.var.followup.fetch_message(self.var.message.id)
+
+        # Determine the emoji to use based on the number of options
+        num = str(message.embeds[0].description).count("\n\n")
+        message.embeds[0].description += f"\n\n{str(self.reactions[num])}  {self.name}"
+
+        await message.edit(embed=message.embeds[0])
+        await interaction.response.defer()
+
+
+class PollButtons(ui.View):
+    def __init__(self, *, timeout=180):
+        super().__init__(timeout=timeout)
+
+    @property
+    def reactions(self):
+        return {
+            0: "1️⃣",
+            1: "2️⃣",
+            2: "3️⃣",
+            3: "4️⃣",
+            4: "5️⃣",
+            5: "6️⃣",
+            6: "7️⃣",
+            7: "8️⃣",
+            8: "9️⃣",
+            9: "🔟",
+        }
```

Comment:
Now that we are using app commands we could use buttons instead of reactions? 👀 

Or would that look bad with 10 options?
What if we used a dropdown?